### PR TITLE
Computer player

### PR DESCRIPTION
**Pull request by Himalee**

### CODE REVIEW

File: lib/display.rb [hunk 1]
```diff
@@ -28,8 +35,33 @@ def choose_cell
...
+    valid_options = [1, 2, 3]
+    until valid_options.include?(response)
+      choose_game_mode
+      response = receive_integer
+    end
+    response
+  end
+
+  def valid_play_again_response
+    response = receive_lower_case_character.downcase
```
Comment (pelensky) [inline lib/display.rb]: `receive_lower_case_character.downcase` indicates the character may not be lower case ;)

File: lib/display.rb [hunk 2]
```diff
@@ -43,4 +75,22 @@ def player_two_wins
...
+    @console.receive.to_i
+  end
+
+  def choose_game_mode
+    @console.present(@message.game_mode)
+  end
+
+  private
+
+  def receive_lower_case_character
```
Comment (pelensky) [inline lib/display.rb]: Is there any validation in this?

File: lib/game_factory.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  def create_game
+    welcome_players
+    present_board
+    players = @player_factory.set_up_players(@game_mode.choose_game_mode, @board, @display)
+    game = Game.new(@display, @board, players)
+  end
+
+  private
+
+  def welcome_players
```
Comment (pelensky) [inline lib/game_factory.rb]: Is this the best class to call this from? Something feels a tad funny to me about having a method responsible for welcoming players in a GameFactory class :)

File: lib/player.rb [hunk 4]
```diff
@@ -1,9 +1,15 @@
...
+    @board = board
+    @display = display
+    @mark = mark
+  end
+
+  def turn
+    @display.choose_cell
+    number = get_cell
+    @board.mark_board(number, @mark)
+    @display.present_board_with_squares(@board)
```
Comment (pelensky) [inline lib/player.rb]: Does the board get printed each turn? Could potentially be called from your runner (I think this is your game class?)

File: lib/player_factory.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+class PlayerFactory
+
+  def set_up_players(choice, board, display)
+    if choice == 1
+      [HumanPlayer.new(board, display, Peg::PLAYER_ONE_MARK), HumanPlayer.new(board, display, Peg::PLAYER_TWO_MARK)]
```
Comment (pelensky) [inline lib/player_factory.rb]: I think we were going to rename Peg to Mark :)

File: spec/computer_player_spec.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  it "returns number 5 given marked board" do
+    @board.mark_board(1, "X")
+    @board.mark_board(2, "X")
+    @board.mark_board(3, "X")
+    @board.mark_board(4, "X")
+    @board.mark_board(6, "X")
+    @board.mark_board(7, "X")
+    @board.mark_board(8, "X")
+    @board.mark_board(9, "X")
+    expect(@computer_player.get_cell).to eql(5)
```
Comment (pelensky) [inline spec/computer_player_spec.rb]: 👍

File: lib/display.rb (hunk 2 above)
Comment (Himalee) [reply lib/display.rb]: @pelensky do you mean like a test?

Comment (Himalee) [reply lib/display.rb]: all good!

**Pull request merged**